magda-web-client/src/Components/Dataset/Add/DatasetAddCommon.ts

```diff
@@ -1234,20 +1230,59 @@ async function convertStateToDistributionRecords(
     return distributionRecords;
 }
 
+/**
+ * Update ckan export aspect status acccording to UI status.
+ * We use JSON Patch request to avoid edging cases
```

Comment:
Might need to mention what the edge case is

Reply:
Add more comments re: edge case